source/docs/flow_control.md

```diff
@@ -87,7 +87,14 @@ the downstream conection.  It passes events to the router filter via
 upstream stream.  The API for provideWatermarkCallbacks is a performance
 optimization to avoid each watermark event on a downstream HTTP/2 connection resulting in
 "number of streams * number of filters" callbacks.  Instead, only the router
-filter is notified and only the "number of streams" multiplier applies.
+filter is notified and only the "number of streams" multiplier applies.  Because
+the router filter only subscribes to notifications when it has an upstream
+connection, the connection manager tracks how many outstanding high watermark
+events have occurred and passes any on to the router filter when it subscribes.
+
+It is worth noting that the router does not unwind `readDisable(true)` calls on
+destruction.  The codecs are responsible for tracking that a given stream no
+longer contributes to the read blocked state and unwinding.
```

Comment:
No need to modify PR, but can you explain how this unwinding is happening above?